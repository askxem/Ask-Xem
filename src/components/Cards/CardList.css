ul {
    position: relative;
    z-index: 2;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding: 0;
    margin: 0;
    gap: 15px;
    justify-content: center;
}

.list li {
    width: 40%;
}

.list img {
    width: 40%;
}

.rainbow {
    z-index: 1;
    position: fixed;
    right: 0px;
    bottom: 0px;
}

.rainbow img {
    height: 100%; 
    position: fixed;
    right: 0px;
    bottom: 0px;
}