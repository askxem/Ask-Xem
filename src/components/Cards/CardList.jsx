
export default function CardList() {
  return (
    <div>
      This is the card list.
    </div>
  )
}
