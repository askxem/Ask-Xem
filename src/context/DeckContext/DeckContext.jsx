import { createContext, useState, useContext } from "react";

const DeckContext = createContext()

const DeckProvider = ({ children }) => {
    const [genSeen, setGenSeen] = useState(JSON.parse(localStorage.getItem('genSeen')) || [])
    const [pronSeen, setPronSeen] = useState(JSON.parse(localStorage.getItem('pronSeen')) || [])

    const seen = (id, category) => {

        if(category == 'gender') {
            let genSet = new Set(genSeen)
            genSet.add(id)
            const arr = Array.from(genSet)
            setGenSeen(arr)
            localStorage.setItem('genSEEN', JSON.stringify(arr))
        }
        if(category == 'pronoun') {
            let pronSet = new Set(pronSeen)
            pronSet.add(id)
            const arr = Array.from(pronSet)
            setPronSeen(arr)
            localStorage.setItem('pronSEEN', JSON.stringify(arr))
        }
    }

    return(
        <DeckContext.Provider value={{ genSeen, pronSeen, seen }}>{children}</DeckContext.Provider>
    )

}

const useDeck = () => {
    const context = useContext(DeckContext)
    if (context === undefined) {
        throw new Error('useDeck not used inside provider')
    }
    return context
}

export { DeckProvider, useDeck }