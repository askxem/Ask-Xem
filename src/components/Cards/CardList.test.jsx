import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router';
import { DeckProvider } from '../../context/DeckContext/DeckContext'
import CardList from './CardList';

const characterList = [{
   id: 1,
   title: 'Pepe Le Pew',
   animal: 'skunk',
   image:'skunkURL',
   category: 'pronoun' 
},
{
    id: 2,
    title: 'Taz',
    animal: 'tazmanian',
    image: 'tazURL',
    category: 'pronoun',
},
{
    id: 3,
    title: 'Tweety',
    animal: 'bird',
    image: 'birdURL',
    category: 'pronoun'
}]

const colors = ['red','orange','yellow']

it('renders front card component', () => {
    const { container } = render(
        <DeckProvider>
            <MemoryRouter>
            <CardList cards={characterList} rainbow={colors}/>
            </MemoryRouter>
        </DeckProvider>
    );

    expect(container).toMatchSnapshot();
})