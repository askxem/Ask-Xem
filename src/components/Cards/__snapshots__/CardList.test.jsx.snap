// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders front card component 1`] = `
<div>
  <ul
    aria-label="cards"
    class="list"
  >
    <li
      class="item"
      style="transform: scale(0) translateZ(0);"
    >
      <a
        href="/pronoun/1"
      >
        <figure
          class="frontcontainer"
        >
          <h3>
            Pepe Le Pew
          </h3>
          <img
            alt="skunk"
            class="image"
            src="/icons/skunk.png"
          />
        </figure>
      </a>
    </li>
    <li
      class="item"
      style="transform: scale(0) translateZ(0);"
    >
      <a
        href="/pronoun/2"
      >
        <figure
          class="frontcontainer"
        >
          <h3>
            Taz
          </h3>
          <img
            alt="tazmanian"
            class="image"
            src="/icons/tazmanian.png"
          />
        </figure>
      </a>
    </li>
    <li
      class="item"
      style="transform: scale(0) translateZ(0);"
    >
      <a
        href="/pronoun/3"
      >
        <figure
          class="frontcontainer"
        >
          <h3>
            Tweety
          </h3>
          <img
            alt="bird"
            class="image"
            src="/icons/bird.png"
          />
        </figure>
      </a>
    </li>
  </ul>
  <div
    class="rainbow"
  >
    <img
      alt="red"
      src="/rainbow/red.png"
      style="opacity: 0;"
    />
    <img
      alt="orange"
      src="/rainbow/orange.png"
      style="opacity: 0;"
    />
    <img
      alt="yellow"
      src="/rainbow/yellow.png"
      style="opacity: 0;"
    />
  </div>
</div>
`;
